geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/sortedset/ZAddExecutor.java

```diff
@@ -46,11 +45,17 @@ public RedisResponse executeCommand(Command command, ExecutionHandlerContext con
       return RedisResponse.error(zAddExecutorState.exceptionMessage);
     }
 
-    long retVal = redisSortedSetCommands.zadd(command.getKey(),
+    Object retVal = redisSortedSetCommands.zadd(command.getKey(),
         new ArrayList<>(commandElements.subList(optionsFoundCount + 2, commandElements.size())),
         makeOptions(zAddExecutorState));
+    if (zAddExecutorState.incrFound) {
+      if (retVal == null) {
+        return RedisResponse.nil();
+      }
+      return RedisResponse.string((byte[]) retVal);
```

Comment:
The [documentation for ZADD with INCR option](https://redis.io/commands/ZADD#return-value) states that the response will be a Bulk string response, not a simple string. I think this should be therefore be `RedisResponse.bulkString(retVal);` The documentation for ZINCRBY also says the same, so I think we may need to change the `RedisResponse` type in the `ZIncrByExecutor` as well, since it's also currently just using a simple string response.

Reply:
Good catch!  I wonder if there are other places where we're returning simple string instead of bulk string...

Reply:
Do we know if there are any (possibly currently disabled) tests in the `redis-test-openjdk11` precheckin job that would fail due to wrong return format?

Reply:
I'm not sure that there are.  I ran a few of the zset.tcl tests against the current implementation (`test "ZINCRBY return value"`, `test "ZINCRBY - increment and decrement`, and `test "ZINCRBY - can create a new sorted set`.. and a few others), but they all passed.

The difference is obvious on the CLI, here's a bulk string returned from native Redis:
```
127.0.0.1:6379> zadd a incr ch 1 a
"3"
```

Here's a simple string returned by us:
```
127.0.0.1:6380> zadd a incr 1 a
1
```
We could write tests at a lower level, but I'm not sure if it's super valuable given that clients are able to handle either one in the same fashion.